components/brave_wallet/browser/zcash/zcash_action_context.cc

```diff
@@ -0,0 +1,31 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_wallet/browser/zcash/zcash_action_context.h"
+
+namespace brave_wallet {
+
+ZCashActionContext::ZCashActionContext(
+    ZCashRpc& zcash_rpc,
+#if BUILDFLAG(ENABLE_ORCHARD)
+    base::SequenceBound<OrchardSyncState>& sync_state,
+#endif  // BUILDFLAG(ENABLE_ORCHARD)
+    const mojom::AccountIdPtr& account_id,
+    const std::string& chain_id)
+    : zcash_rpc(zcash_rpc),
+#if BUILDFLAG(ENABLE_ORCHARD)
+      sync_state(sync_state),
+#endif  // BUILDFLAG(ENABLE_ORCHARD)
```

Comment:
Why are these parameters behind build flags? Also, would it make sense to group them at the end so that they don't need to be repeated twice here?

Reply:
>Why are these parameters behind build flags?

We don't need sync_state for transparent tx support(android, ios cases)

>Also, would it make sense to group them at the end so that they don't need to be repeated twice here?

I think there will be 2 #if blocks anyway